lib/ffmprb/process.rb

```diff
@@ -5,43 +5,50 @@ module Ffmprb
 
   class Process
 
-    def self.intermediate_channel_extname(*media)
-      if media == [:video]
-        '.y4m'
-      elsif media == [:audio]
-        '.wav'
-      elsif media.sort == [:audio, :video]
-        '.flv'
-      else
-        raise Error, "I don't know how to channel [#{media.join ', '}]"
+    class << self
+
+      attr_accessor :duck_audio_hi, :duck_audio_lo
+      attr_accessor :duck_audio_silent_min_sec, :duck_audio_transition_sec
+
+      def intermediate_channel_extname(*media)
+        if media == [:video]
+          '.y4m'
+        elsif media == [:audio]
+          '.wav'
+        elsif media.sort == [:audio, :video]
+          '.flv'
+        else
+          raise Error, "I don't know how to channel [#{media.join ', '}]"
+        end
       end
-    end
 
-    def self.duck_audio(av_main_i, a_overlay_i, silence, av_main_o, video: {resolution: Ffmprb::QVGA, fps: 30})
-      Ffmprb.process(av_main_i, a_overlay_i, silence, av_main_o) do |main_input, overlay_input, duck_data, main_output|
-
-        in_main = input(main_input, **(video ? {} : {only: :audio}))
-        in_over = input(overlay_input, only: :audio)
-        prev_silent_at = 0
-        output(main_output, **(video ? {resolution: video[:resolution], fps: video[:fps]} : {})) do
-          roll in_main
-          ducked_overlay_volume = {0.0 => 0.1}
-          duck_data.each do |silent|
-            next  if silent.end_at && silent.start_at && (silent.end_at - silent.start_at) < 3
-            ducked_overlay_volume.merge!(
-              (silent.start_at - 0.5) => 0.1,
-              (silent.start_at + 0.5) => 0.9
-            )  if silent.start_at
-            ducked_overlay_volume.merge!(
-              (silent.end_at - 0.5) => 0.9,
-              (silent.end_at + 0.5) => 0.1
-            )  if silent.end_at
+      def duck_audio(av_main_i, a_overlay_i, silence, av_main_o, video: {resolution: Ffmprb::QVGA, fps: 30})
```

Comment:
A bit strange that duck_audio is caring about the video resolution, are you sure it should be here?